Sources/OVP/Provider/OVPMediaProvider.swift

```diff
@@ -406,7 +406,21 @@ public enum OVPMediaProviderError: PKError {
                 .set(ks: ks)
             playURL = sourceBuilder.build()
         } else {
-            playURL = source.url
+            if let ks = ks, !ks.isEmpty, let sourceUrl = source.url {
+                if sourceUrl.query == nil {
+                    let lastPathComponent = sourceUrl.lastPathComponent
+                    
+                    playURL = sourceUrl
+                        .deletingLastPathComponent()
+                        .appendingPathComponent("ks")
+                        .appendingPathComponent(ks)
+                        .appendingPathComponent(lastPathComponent)
+                } else {
+                    playURL = sourceUrl.appendingQueryComponent(key: "ks", value: ks)
```

Comment:
Before merging need to set the dependency in the podspec to the Netkit once released.

Reply:
Done